.github/workflows/dbld-images.yml

```diff
@@ -0,0 +1,43 @@
+name: Compile dbld-images
+
+on:
+  pull_request:
+  push:
+    paths:
+      - .github/workflows/dbld-images.yml
+      # List is from dbld/rules cache-image-% "WATCHED_FILES"
+      - "dbld/**"
+      - "packaging/rhel/syslog-ng.spec"
+      - "packaging/debian/control"
+  schedule:
+    - cron: '00 03 * * *'
+  workflow_dispatch:
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+  
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v2
+
+      - name: Build the images
+        run: dbld/rules images -j $(nproc)
+
+      - name: Log in to the Container registry
+        if: github.repository_owner == 'syslog-ng' && github.ref == 'refs/heads/master'
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Push the images
+        if: github.repository_owner == 'syslog-ng' && github.ref == 'refs/heads/master'
```

Comment:
Shouldn't we add ` github.event_name == 'push'` as well? (we have a cron schedule here)

Reply:
While the code do not express it properly, it was intentional from my side to update the images during the nightly build. So upstream package changes in the different distributions can flow in.

I think, Your comment raises awareness for an existing issue: Two images can be different even with the same `COMMIT` label.

Reply:
Hm, okay. I hope this won't break too frequently (but at least we'll know sooner than before).

My understanding is that `COMMIT` is just a label that shows the "version" of the Dockerfile for the given image, so I think nothing is wrong with the fact that `COMMIT` does not uniquely identify a docker image (docker image IDs and tags are for this).

Reply:
Thinking it over again. We can disable the image upload on the nightly run.

- The nightly job will still fail in the build process. So we will notice if something goes wrong.
- We can steal some time with it, because the cached images will allow us to merge to the master.
- When we correct it, the "path" restriction (similar to the dbld's cache-image) will trigger the new image build.

Reply:
Sounds great, let's do that.